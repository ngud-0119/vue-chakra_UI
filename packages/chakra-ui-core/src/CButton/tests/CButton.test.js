import { CButton } from '../..'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const base = {
    components: {
      CButton
    },
    template: '<CButton>Works</CButton>',
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should display children', () => {
  const { getByText } = renderComponent({ template: '<CButton><span>Works</span></CButton>' })
  expect(getByText('Works')).toBeInTheDocument()
})

it('should display button with left icon', () => {
  const { container, asFragment } = renderComponent({ template: '<CButton leftIcon="email">Email</CButton>' })
  expect(container.querySelector('button > svg')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

it('should display button with right icon', () => {
  const { container, asFragment } = renderComponent({ template: '<CButton rightIcon="email">Email</CButton>' })
  expect(container.querySelector('button > svg')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

it('should display spinner and hide text', () => {
  const { getByTestId, container } = renderComponent({ template: '<CButton isLoading data-testid="btn">CButton</CButton>' })
  const button = getByTestId('btn')

  expect(button).toHaveAttribute('disabled')
  expect(button).toHaveAttribute('aria-disabled', 'true')

  const spinner = container.querySelector('[data-chakra-component=CSpinner]')
  expect(spinner).toBeInTheDocument()
  expect(button).toHaveStyle('opacity: 0.4')
})

it('should display spinner with text', () => {
  const { getByText, container } = renderComponent({ template: '<CButton isLoading loadingText="Submitting" data-testid="Spinner">Button</CButton>' })

  expect(getByText('Submitting')).toBeInTheDocument()
  const spinner = container.querySelector('[data-chakra-component=CSpinner]')
  expect(spinner).toBeInTheDocument()
})

it('should display a disabled button', () => {
  const { getByText } = renderComponent({ template: '<CButton isDisabled>Button</CButton>' })

  expect(getByText('Button')).toHaveAttribute('disabled')
})
