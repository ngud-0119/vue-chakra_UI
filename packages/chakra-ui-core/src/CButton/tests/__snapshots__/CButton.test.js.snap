// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButton should display button with left icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  font-weight: var(--chakra-fontWeights-bold);
  border-radius: var(--chakra-radii-md);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-gray-100);
}

.emotion-0:focus,
.emotion-0[data-focus] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-gray-200);
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-1:not(:root) {
  overflow: hidden;
}

.emotion-2 {
  width: 1em;
  height: 1em;
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 {
  color: currentColor;
  width: 1em;
  height: 1em;
}

.emotion-4 {
  margin-right: var(--chakra-space-1);
  margin-bottom: var(--chakra-space-px);
  opacity: 1;
}

<button
    class="emotion-0"
    data-chakra-component="CButton"
    tabindex="0"
    type="button"
  >
    <svg
      class="emotion-1 emotion-2 emotion-3 emotion-4"
      data-chakra-component="CButtonIcon"
      icon="email"
      role="presentation"
      viewBox="0 0 24 24"
    >
      
    
      <g
        fill="currentColor"
      >
        
      
        <path
          d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
        />
        
      
        <path
          d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
        />
        
    
      </g>
      
    
    </svg>
    Email
  </button>
</DocumentFragment>
`;

exports[`CButton should display button with right icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  font-weight: var(--chakra-fontWeights-bold);
  border-radius: var(--chakra-radii-md);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-gray-100);
}

.emotion-0:focus,
.emotion-0[data-focus] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-gray-200);
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-1:not(:root) {
  overflow: hidden;
}

.emotion-2 {
  width: 1em;
  height: 1em;
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 {
  color: currentColor;
  width: 1em;
  height: 1em;
}

.emotion-4 {
  margin-left: var(--chakra-space-1);
  margin-bottom: var(--chakra-space-px);
  opacity: 1;
}

<button
    class="emotion-0"
    data-chakra-component="CButton"
    tabindex="0"
    type="button"
  >
    Email
    <svg
      class="emotion-1 emotion-2 emotion-3 emotion-4"
      data-chakra-component="CButtonIcon"
      role="presentation"
      viewBox="0 0 24 24"
    >
      
    
      <g
        fill="currentColor"
      >
        
      
        <path
          d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
        />
        
      
        <path
          d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
        />
        
    
      </g>
      
    
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CButton should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  font-weight: var(--chakra-fontWeights-bold);
  border-radius: var(--chakra-radii-md);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-gray-100);
}

.emotion-0:focus,
.emotion-0[data-focus] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-gray-200);
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

<button
    class="emotion-0"
    data-chakra-component="CButton"
    tabindex="0"
    type="button"
  >
    Works
  </button>
</DocumentFragment>
`;
